### ISSUE TITLE
Does it Work with Tibia 11?

### ISSUE THREAD

**Issue by oufooz** [Mytherin/Tibialyzer#137]
does it work with tibia 11 ?      or will it be developed for it ?

**Comment by KuerbisGnu** [comment]
Mytherin:
>  The beta client has a different internal structure than the regular client, and Tibialyzer will have to be patched to work with it.
> 
> I have done some digging into the memory of the client and have found that strings are stored as follows.
> 
> struct {
>     short int length;    // 2 bytes indicate length of string
>     char string[length]; // the actual string
> };
> The strings aren't null-terminated, thus the length must be used to figure out when the strings end.
> 
> The main problem is that log messages are split up into several separate strings. Chat messages, for example, are split up into the separate strings Name and Message (possibly also Time and Level).
> 
> That means that to get a complete log message we need to understand how these different strings are combined, which means we need to reverse engineer the structure of the server log. I haven't yet found out how they are tied together. I will need to dig deeper later.

**Comment by oufooz** [comment]
Can u somehow encapsulate that part of the problem if u figure it out.  so that its not visible anymore.. so that botters have a hard-time figuring out how to come back after that update?.

**Comment by Mytherin** [comment]
Botters need to figure out something entirely different from what I need to figure out. I don’t think bots even use the chat log at all. They need to figure out how to find the exact positions of creatures, their life, how many/which creatures there are, the position of the player in the world, etc. They don’t really need the chat log for botting, while that’s the only thing I need for Tibialyzer.

**Comment by oufooz** [comment]
Stuff like heals bots.. can use the chat log. a fully functional bot is a different thing.

**Comment by mooop12** [comment]
Don't mean to rush, but has there been any progress to this? Since Tibia 11 is now in open beta I'd imagine it's not too far away from release.

**Comment by Mytherin** [comment]
Yes, I have done some additional exploratory work on Tibia 11. I now know how to read the chat logs but haven't looked into other stats yet (life/mana/etc), and I still need to implement it in Tibialyzer as well. When Tibia 11 releases I will release a Tibialyzer that works with it unless CIP massively overhauls the internal structures, although I can't promise that all features will work with it (e.g. HUDs).

**Comment by gabrielrdrguez** [comment]
@Mytherin  Could you release an unsupported version to Tibia 11 with only looting stuff? I mean, that was originally the first step for tibialyzer and I guess it's the most relevant function until today. Some feedback could also help you when Tibia 11 finally arrives

**Comment by Mytherin** [comment]
@gabrielrdrguez [Done](https://github.com/Mytherin/Tibialyzer/releases/tag/1.4.0-beta).

**Comment by gabrielrdrguez** [comment]
@Mytherin Thanks man! You're awesome, I gonna try it and give you all the the feedback that I can provide.

**Comment by brocades** [comment]
Trying to use the version 1.4.2 with Tibia 11 but I just can't find a way to see the summary window while playing =/.
